Allow disabling compression

kamalmarhubi opened the pull request:
This adds a `--compression` flag which can take the values `gzip` and
`none`. It is enum-like rather than boolean to allow xz and bzip2 to be
added later.

Refs #110

iaguis commented on lib/common/common.go:
```diff
@@ -33,6 +33,13 @@ const (
+
+const (
+	NoCompression = iota
+	GzipCompression = iota
```
You don't need this `iota`

kamalmarhubi replied:
Cargo culting... I don't really know go! :p

Done.

iaguis commented on lib/docker2aci.go:
```diff
@@ -224,10 +229,19 @@ func getManifests(renderedACI acirenderer.RenderedACI, aciRegistry acirenderer.A
+		tarWriterTarget = gzip.NewWriter(outputFile)
+		defer tarWriterTarget.Close()
+	default:
+		return fmt.Errorf("Unexpected compression enum value: %d", compression)
```
we should return lower-cased errors

kamalmarhubi replied:
Done.

iaguis commented on lib/common/common.go:
```diff
@@ -358,7 +365,7 @@ func writeACI(layer io.ReadSeeker, manifest schema.ImageManifest, curPwl []strin
 
 	var w io.WriteCloser = aciFile
-	if compress {
+	if compression != NoCompression {
```
Shouldn't we check specifically for gzip here?

kamalmarhubi replied:
Done.

iaguis commented:
Thanks for the PR! I've made some comments but looks good :)

kamalmarhubi commented:
@iaguis thanks for the quick response! As an operational note for addressing the comments, do you prefer commits to be added, or to rewrite the commit?

kamalmarhubi commented:
I've pushed them as added commits. Let me know if you want me to squash before you merge!

iaguis commented:
Yeah, please squash them and then I'll merge it. Thanks a lot!

kamalmarhubi commented:
All done! Thanks again for the quick review. I'm tempted to put in xz and bzip2 compression (using installed utilities if available) base on that alone!
